frontend/src/lib/components/heatmaps/heatmapDataLogic.ts

```diff
@@ -37,6 +36,7 @@ export const heatmapDataLogic = kea<heatmapDataLogicType>([
         setHeatmapFixedPositionMode: (mode: HeatmapFixedPositionMode) => ({ mode }),
         setHeatmapColorPalette: (Palette: string | null) => ({ Palette }),
         setHref: (href: string) => ({ href }),
+        setHrefMatchType: (matchType: 'exact' | 'pattern') => ({ matchType }),
```

Comment:
the data logic is used in two different ways, so we need to make the match type explicit